crates/wasm-metadata/src/lib.rs

```diff
@@ -672,8 +747,344 @@ fn indirect_name_map(
+        self.license.as_ref()
+    }
+
+    /// Set license
+    pub fn set_license(&mut self, license: Option<String>) {
+        self.license = license;
+    }
+
+    /// Get custom_licenses
+    pub fn get_custom_licenses(&self) -> Option<&Vec<CustomLicense>> {
+        self.custom_licenses.as_ref()
+    }
+
+    /// Set custom_licenses
+    pub fn set_custom_licenses(&mut self, custom_licenses: Option<Vec<CustomLicense>>) {
+        self.custom_licenses = custom_licenses;
+    }
+
+    /// Get links
+    pub fn get_links(&self) -> Option<&Vec<Link>> {
+        self.links.as_ref()
+    }
+
+    /// Set links
+    pub fn set_links(&mut self, links: Option<Vec<Link>>) {
+        self.links = links;
+    }
+
+    /// Get categories
+    pub fn get_categories(&self) -> Option<&Vec<String>> {
+        self.categories.as_ref()
+    }
+
+    /// Set categories
+    pub fn set_categories(&mut self, categories: Option<Vec<String>>) {
+        self.categories = categories;
+    }
+
+    fn display(&self, f: &mut fmt::Formatter, indent: usize) -> fmt::Result {
+        let spaces = std::iter::repeat(" ").take(indent).collect::<String>();
+
+        if let Some(authors) = &self.authors {
+            writeln!(f, "{spaces}authors:")?;
+            for author in authors {
+                writeln!(f, "{spaces}    {author}")?;
+            }
+        }
+
+        if let Some(license) = &self.license {
+            writeln!(f, "{spaces}license:")?;
+            writeln!(f, "{spaces}    {license}")?;
+        }
+
+        if let Some(links) = &self.links {
+            writeln!(f, "{spaces}links:")?;
+            for link in links {
+                writeln!(f, "{spaces}    {link}")?;
+            }
+        }
+
+        if let Some(categories) = &self.categories {
+            writeln!(f, "{spaces}categories:")?;
+            for category in categories {
+                writeln!(f, "{spaces}    {category}")?;
+            }
+        }
+
+        if let Some(description) = &self.description {
+            writeln!(f, "{spaces}description:")?;
+            writeln!(f, "{spaces}    {description}")?;
+        }
+
+        if let Some(custom_licenses) = &self.custom_licenses {
+            writeln!(f, "{spaces}custom_licenses:")?;
+            for license in custom_licenses {
+                license.display(f, indent + 4)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Display for RegistryMetadata {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.display(f, 0)
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub struct Link {
+    pub ty: LinkType,
+    pub value: String,
+}
+
+impl Display for Link {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}: {}", self.ty, self.value)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub enum LinkType {
+    Documentation,
+    Homepage,
+    Repository,
+    Funding,
+    #[serde(untagged)]
```

Comment:
Is this a valid serde field attribute? Doesn't it belong on the enum declaration?

Reply:
Oddly enough, my build was failing locally because of this, but no longer does with a clean build. Perhaps something changed in serde to support it on the variant?

Reply:
Oh if you're like me and your `Cargo.lock` was out of date (probably by a year or so for me) it was fixed by a `cargo update`, so I think we may need to update the `serde` dependency version requirement to handle this

Reply:
Yep, looks like a [change to serde](https://github.com/serde-rs/serde/pull/2403) that the documentation hasn't caught up with that released in a non-semver-compatible way (hence why my build was broken, but a newer patch version of `serde` fixed it).